lib/ruby_lsp_rails/middleware.rb

```diff
@@ -0,0 +1,63 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Rails
+    class Middleware
```

Comment:
We might later want to consider extracting some of this into `ruby-lsp`, so there's less 'boilerplate' necessary for extension authors.

Reply:
But if we won't want them to do this, what's the point of extracting them to `ruby-lsp`?

This is Rack/Rails specific pattern and not what `ruby-lsp` should have provided?

Reply:
I agree with Stan here, `ruby-lsp-rails` will be quite a unique extension since it will be communicating with a live server. I suspect most other extensions wouldn't need it - and we shouldn't create incentive for people to do so as getting this wrong can be detrimental to the overall experience.

Reply:
I think there there could be other neat extensions powered by a similar mechanism, but I agree with not trying to optimize for that until we actually have more real-world examples.